src/ngx_http_lua_regex.c

```diff
@@ -1922,12 +1924,78 @@ ngx_http_lua_ngx_re_sub_helper(lua_State *L, unsigned global)
 }
 
 
+ngx_int_t
+ngx_http_lua_set_jit_stack_size(int size)
+{
+    ngx_http_lua_main_conf_t    *lmcf;
+
+    lmcf = ngx_http_cycle_get_module_main_conf(ngx_cycle,
+                                               ngx_http_lua_module);
+
+    if (lmcf->regex_cache_entries > 0) {
+        return -1;
```

Comment:
Better avoid using magical numbers like `-1` and `-2` here. We could reuse existing constants like `NGX_ERROR`, `NGX_DECLINED` and `NGX_BUSY` here, if applicable.